frontend/public/components/secrets/create-secret/PullSecretCredentialsForm.tsx

```diff
@@ -0,0 +1,151 @@
+import { useEffect, useState } from 'react';
+
+export const PullSecretCredentialsForm: React.FC<PullSecretCredentialsFormProps> = ({
+  onChange,
+  stringData,
+}) => {
+  const { t } = useTranslation();
+  const isDockerconfigjson = _.isEmpty(stringData) || !!stringData[AUTHS_KEY];
+  const newImageSecretEntry = (): SecretEntry => ({
+    entry: {
+      address: '',
+      username: '',
+      password: '',
+      email: '',
+      auth: '',
+    },
+    uid: _.uniqueId(),
+  });
+  const imageSecretObjectToArray = (imageSecretObject): SecretEntry[] => {
+    const imageSecretArray = [];
+    if (_.isEmpty(imageSecretObject)) {
+      return _.concat(imageSecretArray, newImageSecretEntry());
+    }
+    _.each(imageSecretObject, (v, k) => {
+      // Decode and parse 'auth' in case 'username' and 'password' are not part of the secret.
+      const decodedAuth = Base64.decode(_.get(v, 'auth', ''));
+      const parsedAuth = _.isEmpty(decodedAuth) ? _.fill(Array(2), '') : _.split(decodedAuth, ':');
+      imageSecretArray.push({
+        entry: {
+          address: k,
+          username: _.get(v, 'username', parsedAuth[0]),
+          password: _.get(v, 'password', parsedAuth[1]),
+          email: _.get(v, 'email', ''),
+          auth: _.get(v, 'auth', ''),
+        },
+        uid: _.get(v, 'uid', _.uniqueId()),
+      });
+    });
+    return imageSecretArray;
+  };
+  const imageSecretArrayToObject = (imageSecretArray: SecretEntry[]) => {
+    const imageSecretsObject = {};
+    _.each(imageSecretArray, (value) => {
+      imageSecretsObject[value.entry.address] = _.pick(value.entry, [
+        'username',
+        'password',
+        'auth',
+        'email',
+      ]);
+    });
+    return imageSecretsObject;
+  };
+
+  const [secretEntriesArray, setSecretEntriesArray] = useState(
+    imageSecretObjectToArray(stringData?.[AUTHS_KEY] || stringData),
+  );
+  const propogateEntryChange = (secretEntries: SecretEntry[]) => {
+    const imageSecretObject = imageSecretArrayToObject(secretEntries);
+    onChange(isDockerconfigjson ? { [AUTHS_KEY]: imageSecretObject } : imageSecretObject);
+  };
+
+  useEffect(() => {
+    propogateEntryChange(secretEntriesArray);
+  }, [secretEntriesArray]);
+
+  const onDataChanged = (updatedEntry: Entry, entryIndex: number) => {
+    const updatedEntryData = {
+      uid: secretEntriesArray[entryIndex].uid,
+      entry: updatedEntry,
+    };
+    const updatedSecretEntriesArray = [...secretEntriesArray];
+    updatedSecretEntriesArray[entryIndex] = updatedEntryData;
+    setSecretEntriesArray(updatedSecretEntriesArray);
+  };
+
+  const removeEntry = (entryIndex: number) => {
+    const updatedSecretEntriesArray = [...secretEntriesArray];
+    updatedSecretEntriesArray.splice(entryIndex, 1);
+    setSecretEntriesArray(updatedSecretEntriesArray);
+  };
+
+  const addEntry = () => {
+    setSecretEntriesArray(_.concat(secretEntriesArray, newImageSecretEntry()));
+  };
```

Comment:
We shouldn't need state in this component. Entries are memo-ized as recommended, then each time a change occurs in the form fields, we create an updated entry array, transform it to pull secret data, and pass it to the onChange prop. Also, our callbacks should have a more cohesive naming "scheme" for clarity. For example:
```ts
const onEntriesChanged = (entries) => { // was "propagateEntryChange" }
const updateEntry = (entry, atIndex) => { // was "onDataChanged" }
const removeEntry = (atIndex) => { }
const addEntry = () => { }